var/spack/repos/builtin/packages/quantum-espresso/package.py

```diff
@@ -79,6 +75,37 @@ class QuantumEspresso(CMakePackage):
         conflicts("^openblas threads=none")
         conflicts("^openblas threads=pthreads")
 
+    # Add cuda support
+    with when("%nvhpc"):
+        variant(
+            "cuda",
+            default=False,
+            description="Build with CUDA, OpenMP on is strongly suggested for performance",
+        )
+        with when("+cuda"):
+            # GPUs are enabled since v6.6
+            conflicts("@:6.5")
+            # cuda version >= 10.1
+            # conflicts("cuda@:10.0.130")
+            # bugs with following nvhpcsdk versions
+            conflicts(
+                "%nvhpc@21.11:22.2", msg="QE-GPU has issues with these NVHPC compiler versions"
+            )
+            # bugs with nvhpc@22.3 in versions > 7.1 still present
+            with when("@develop"):
+                conflicts("%nvhpc@22.3", msg="Develop has issues with this NVHPC compiler version")
+            with when("@7.2:"):
+                conflicts("%nvhpc@22.3", msg="v > 7.1 has issues with this NVHPC compiler version")
```

Comment:
I think 22.3 works fine after 7.1 release based on https://gitlab.com/QEF/q-e/-/issues/479#note_927308910

Reply:
I tested the develop at commit 77c166c6ebb13ea0bf13d846bcd6c812e2584825 with NVHPC/22.3  CUDA/11.5 OpenMPI/4.1.2 on juwels booster and it does not converge with Davidson diagonalization

Reply:
Got it. Probably we should just black list 22.3 completely.

Reply:
done in 487d76f99948010ddf0daf8cd1eaf35a2dd9929e